sale_by_packaging/tests/test_packaging_type_can_be_sold.py

```diff
@@ -38,19 +38,44 @@ def setUpClass(cls):
                 "packaging_type_id": cls.packaging_type_cannot_be_sold.id,
             }
         )
-
-    def test_packaging_type_can_be_sold(self):
-        order = self.env["sale.order"].create({"partner_id": self.partner.id})
-        order_line = self.env["sale.order.line"].create(
+        cls.order = cls.env["sale.order"].create({"partner_id": cls.partner.id})
+        cls.order_line = cls.env["sale.order.line"].create(
             {
-                "order_id": order.id,
-                "product_id": self.product.id,
-                "product_uom": self.product.uom_id.id,
+                "order_id": cls.order.id,
+                "product_id": cls.product.id,
+                "product_uom": cls.product.uom_id.id,
                 "product_uom_qty": 3.0,
             }
         )
-        order_line.write({"product_packaging": self.packaging_can_be_sold.id})
+
+    def test_packaging_type_can_be_sold(self):
+        self.order_line.write({"product_packaging": self.packaging_can_be_sold.id})
         with self.assertRaises(ValidationError):
-            order_line.write({"product_packaging": self.packaging_cannot_be_sold.id})
-            onchange_res = order_line._onchange_product_packaging()
+            self.order_line.write(
+                {"product_packaging": self.packaging_cannot_be_sold.id}
+            )
+            onchange_res = self.order_line._onchange_product_packaging()
             self.assertIn("warning", onchange_res)
+
+    def test_product_packaging_can_be_sold(self):
+        """Check that a product.packaging can be independently set as can be sold.
+        """
+        with self.assertRaises(ValidationError):
```

Comment:
can you test which exception msg you get here? You can use `with self.assertRaisesRegex(TypeError, msg)`

Reply:
Done